### PR TITLE
[Kotlin Migration] HelpDialog

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
```
> Task :AnkiDroid:compilePlayReleaseKotlin
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ImportFileSelectionFragment.kt: (36, 47): Interface ActivityConsumer does not have constructors
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ImportFileSelectionFragment.kt: (36, 66): Cannot infer a type for this parameter. Please specify it explicitly.
```

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/HelpDialog.kt [hunk 1]
```diff
@@ -13,343 +13,325 @@
...
+                }
             }
-        };
+        }
     }
 
-    public static class FunctionItem extends Item implements Parcelable {
-        private final ActivityConsumer mFunc;
+    class FunctionItem : RecursivePictureMenu.Item, Parcelable {
+        private val mFunc: ActivityConsumer?
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/HelpDialog.kt:205]: `@KotlinCleanup`: `lateinit var`

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/HelpDialog.kt [hunk 2]
```diff
@@ -13,343 +13,325 @@
...
-            @Override
-            public FunctionItem[] newArray(int size) {
-                return new FunctionItem[size];
-            }
-        };
-
         @FunctionalInterface
-        public interface ActivityConsumer extends Serializable {
-            void consume(@NonNull AnkiActivity activity);
+        interface ActivityConsumer : Serializable {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/HelpDialog.kt:229]: `@KotlinCleanup`: Change to Kotlin Functional Interface

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
small nits

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]
nice! That's a big one

**Pull request merged**